### PR TITLE
Handle new package file

### PR DESCRIPTION
**Pull request by tothszabi**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

The e2e tests started to fail on the Linux stack. After some investigation I have found that the Dart team deprecated the way they were handling the packages and now there is a new json based variant:
https://github.com/dart-lang/sdk/issues/48272

They were generously generating the old version too for backwards compatibility but after a certain version it will be completely removed. 

This PR introduces handling this new json format. 

### Changes

The step will first try to parse the old `.packages` file. If it fails then it will also try to parse the new `.dart_tool/package_config.json` one.

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -234,3 +231,63 @@ func cacheFlutterDeps(projectDir string) error {
...
+
+	return packages, nil
+}
+
+func parseJSON(contents string) (map[string]url.URL, error) {
+	type packageConfig struct {
+		Packages []struct {
+			Name       string `json:"name"`
+			RootUri    string `json:"rootUri"`
+			PackageUri string `json:"packageUri"`
```
Comment (AAverin) [inline cache.go:270]: Looking at https://dart.googlesource.com/sdk/+/5ae183856945440597e0674436dd89b9e08632bc/.dart_tool/package_config.json as an example, at least `packageUri` can be missing. Should this field rather be `string?` and `omitempty`?

Comment (vshah23) [reply cache.go:270]: Nice catch!

Comment (tothszabi) [reply cache.go:270]: Go does not have option fields and the way it works is that it initialises every field to the fields default value. For strings this is the `""` empty string. Which is great for us and this way we do not need to care if the `packageUri` exists or not. 

The `omitempty` tag is only useful for encoding. It is not used during decoding.

**Review by AAverin — DISMISSED** [review]

File: cache.go (hunk 1 above)
Comment (AAverin) [reply cache.go:270]: Don't know what is exactly the rule here.
But in my old team we usually approved the code pro-actively allowing the author to fix comments and tests and merge without blockers.
Approved =)

**Review by AAverin — APPROVED** [review]

**Pull request merged**